### PR TITLE
Treat null code fences as unlabelled

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- handle `null` fence languages as absent in fence compression
- attach orphan specifiers when encountering `null` language fences
- document and test fence `null` handling

## Testing
- `make fmt`
- `make lint`
- `make test`
- `make markdownlint`
- `make nixie`

------
https://chatgpt.com/codex/tasks/task_e_68b6a21e95e4832290f7be2a2b024240

## Summary by Sourcery

Treat null code fence languages as unlabeled in both compress_fences and attach_orphan_specifiers, add tests for null-language handling, and update documentation accordingly

New Features:
- Treat 'null' code fence languages as absent/unlabeled in fence compression
- Attach orphan specifiers to code fences labeled 'null'

Documentation:
- Document null-language fence behavior in the architecture documentation

Tests:
- Add unit tests for compressing null-language fences
- Add unit tests for attaching orphan specifiers to null-language fences
- Add CLI integration test for handling null-language fences

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -22,8 +22,9 @@ The function combines several helpers documented in `docs/`:
 
 - `fences::compress_fences` and `attach_orphan_specifiers` normalize code block
   delimiters. The latter keeps indentation from the language line when the
-  fence lacks it. It also tolerates spaces within comma-separated specifiers,
-  e.g. `TOML, Ini` becomes `toml,ini`.
+  fence lacks it. Language specifiers explicitly set to `null` are treated as
+  absent. It also tolerates spaces within comma-separated specifiers, e.g.
+  `TOML, Ini` becomes `toml,ini`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/architecture.md]: @codex has this issue identified in this comment been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: tests/fences.rs [hunk 2]
```diff
@@ -201,3 +201,17 @@ fn attaches_orphan_specifier_allows_spaces() {
     let out = attach_orphan_specifiers(&compress_fences(&input));
     assert_eq!(out, lines_vec!["```toml,ini", "a=1", "```"]);
 }
+
+#[test]
+fn compresses_null_language_to_empty() {
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/fences.rs]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai have the following now been resolved?

- Extract the repeated “is empty or equals null” checks into a small helper (e.g. `is_blank_or_null_lang`) to reduce duplication and clarify intent.
- Since `compress_fences` already normalizes `null` to an empty fence, you could simplify `attach_orphan_specifiers` by only handling empty languages and removing the extra null branches.
- Add a test for uppercase `NULL` (e.g. ```NULL) to ensure the case‐insensitive match behaves as expected.

File: docs/architecture.md [hunk 3]
```diff
@@ -20,10 +20,11 @@ pub fn process_stream_inner(lines: &[String], opts: Options) -> Vec<String>
 
 The function combines several helpers documented in `docs/`:
 
-- `fences::compress_fences` and `attach_orphan_specifiers` normalize code block
+- `fences::compress_fences` and `attach_orphan_specifiers` normalise code block
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/architecture.md]: @codex has this issue identified in this comment been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: docs/architecture.md [hunk 4]
```diff
@@ -20,10 +20,11 @@
...
   delimiters. The latter keeps indentation from the language line when the
-  fence lacks it. It also tolerates spaces within comma-separated specifiers,
-  e.g. `TOML, Ini` becomes `toml,ini`.
+  fence lacks it. Language specifiers explicitly set to `null`
+  (case-insensitive) are treated as absent. It also tolerates spaces within
+  comma-separated specifiers, e.g. `TOML, Ini` becomes `toml,ini`.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/architecture.md]: @codex has this issue identified in this comment been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: tests/cli.rs [hunk 5]
```diff
@@ -303,3 +303,20 @@ fn run_in_place(flags: &[&str], input: &str, expected: &str) {
...
+#[case("~~~~NULL\nfn main() {}\n~~~~\n", "```\nfn main() {}\n```\n")]
+fn test_cli_fences_null_language(#[case] input: &str, #[case] expected: &'static str) {
+    Command::cargo_bin("mdtablefix")
+        .expect("Failed to create cargo command for mdtablefix")
+        .arg("--fences")
+        .write_stdin(input)
+        .assert()
+        .success()
+        .stdout(expected);
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/cli.rs]: @codex has this issue identified in this comment been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

File: tests/fences.rs [hunk 6]
```diff
@@ -201,3 +202,48 @@ fn attaches_orphan_specifier_allows_spaces() {
...
+#[case("~~~~NULL", "~~~~")]
+#[case("~~~~Null", "~~~~")]
+#[case("~~~~null  ", "~~~~")]
+#[case("~~~~NULL  ", "~~~~")]
+#[case("~~~~Null  ", "~~~~")]
+fn compresses_null_language_to_empty(#[case] open: &str, #[case] close: &str) {
+    let input = lines_vec![open, "code", close];
+    let out = compress_fences(&input);
+    assert_eq!(out, lines_vec!["```", "code", "```"]);
+}
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/fences.rs]: @codex has this issue identified in this comment been resolved? Please provide evidence for your assessment.

If work remains, please provide clear self-contained instructions (including rationale and location) in a triple-tilde code-fenced block.

**Comment by leynos** [comment]
@coderabbitai has the following now been resolved?

❌ attach_orphan_specifiers simplification: However, the second point has not been fully addressed. The attach_orphan_specifiers function still explicitly checks for null languages using is_null_lang(lang_present) rather than being simplified to only handle empty languages after compress_fences normalization.

The current implementation still has separate logic branches for null/empty languages in attach_orphan_specifiers, when it could potentially be simplified since compress_fences already normalizes null languages to empty fences. This means there's still some redundancy in the null-handling logic between the two functions.
